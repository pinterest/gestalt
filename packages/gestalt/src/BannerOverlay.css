.textColorOverrideLight * {
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--color-text-dark) !important;
}

.textColorOverrideDark * {
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--color-text-light) !important;
}

.parentButton {
  composes: noBorder from "./Borders.css";
  composes: p0 from "./Whitespace.css";
  composes: paddingX0 from "./boxWhitespace.css";
  composes: paddingY0 from "./boxWhitespace.css";
  composes: transparentBg from "./Colors.css";
  text-wrap: nowrap;
}
