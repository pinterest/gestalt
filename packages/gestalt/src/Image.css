.img {
  composes: block from "./Layout.css";
  composes: xsCol12 from "./Column.css";
  composes: absolute from "./Layout.css";
}

/*
    Disables alt text showing when loading an image in firefox
    https://stackoverflow.com/questions/12667926
*/

.img[alt] {
  color: transparent;
}

.scaled-img {
  composes: relative from "./Layout.css";
  background-position: center center;
  background-repeat: no-repeat;
  height: 100%;
  width: 100%;
}

.contain {
  composes: scaled-img;
  background-size: contain;
}

.cover {
  composes: scaled-img;
  background-size: cover;
}

@media (inverted-colors) {
  .img,
  .scaled-img {
    filter: invert(100%);
  }
}
