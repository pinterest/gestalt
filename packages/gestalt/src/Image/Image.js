// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import Box from '../Box/Box';
import styles from './Image.css';

const shouldScaleImage = fit => fit === 'cover' || fit === 'contain';

type Props = {|
  alt: string,
  children?: React.Node,
  color: string,
  fit: 'contain' | 'cover' | 'none',
  naturalHeight: number,
  naturalWidth: number,
  onError?: () => void,
  onLoad?: () => void,
  sizes?: string,
  src: string,
  srcSet?: string,
|};

export default class Image extends React.PureComponent<Props> {
  static propTypes = {
    alt: PropTypes.string.isRequired,
    children: PropTypes.node,
    color: PropTypes.string,
    fit: PropTypes.oneOf(['contain', 'cover', 'none']),
    naturalHeight: PropTypes.number.isRequired,
    naturalWidth: PropTypes.number.isRequired,
    onError: PropTypes.func,
    onLoad: PropTypes.func,
    sizes: PropTypes.string,
    src: PropTypes.string.isRequired,
    srcSet: PropTypes.string,
  };

  static defaultProps = {
    color: 'transparent',
    fit: 'none',
  };

  componentDidMount() {
    if (shouldScaleImage(this.props.fit)) {
      this.loadImage();
    }
  }

  componentDidUpdate(prevProps: Props) {
    const { fit, src } = this.props;
    if (shouldScaleImage(fit) && prevProps.src !== src) {
      this.loadImage();
    }
  }

  handleLoad = () => {
    if (this.props.onLoad) {
      this.props.onLoad();
    }
  };

  handleError = () => {
    if (this.props.onError) {
      this.props.onError();
    }
  };

  loadImage() {
    if (typeof window !== 'undefined') {
      const image = new window.Image();
      image.onload = this.handleLoad;
      image.onerror = this.handleError;
      image.src = this.props.src;
    }
  }

  render() {
    const {
      alt,
      color,
      children,
      fit,
      naturalHeight,
      naturalWidth,
      sizes,
      src,
      srcSet,
    } = this.props;

    const isScaledImage = shouldScaleImage(fit);
    const childContent = children ? (
      <Box position="absolute" top left bottom right overflow="hidden">
        {children}
      </Box>
    ) : null;

    return isScaledImage ? (
      <div
        aria-label={alt}
        className={styles[fit]}
        style={{
          backgroundColor: color,
          backgroundImage: `url('${src}')`,
        }}
        role="img"
      >
        {childContent}
      </div>
    ) : (
      <Box
        position="relative"
        dangerouslySetInlineStyle={{
          __style: {
            backgroundColor: color,
            paddingBottom: `${naturalHeight / naturalWidth * 100}%`,
          },
        }}
      >
        <img
          alt={alt}
          className={styles.img}
          onError={this.handleError}
          onLoad={this.handleLoad}
          sizes={sizes}
          src={src}
          srcSet={srcSet}
        />
        {childContent}
      </Box>
    );
  }
}
