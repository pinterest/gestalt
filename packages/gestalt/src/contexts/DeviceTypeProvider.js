// @flow strict
import { type Context, createContext, type Node, useContext } from 'react';

const defaultDeviceType = 'desktop';

type DeviceType = 'desktop' | 'mobile';

const DeviceTypeContext: Context<DeviceType> = createContext<DeviceType>(defaultDeviceType);

type Props = {
  children: Node,
  /**
   * The device type as determined by logic within your app.
   */
  deviceType: DeviceType,
};

/**
 * [DeviceTypeProvider](https://gestalt.pinterest.systems/web/utilities/devicetypeprovider) is an optional [React Context provider](https://reactjs.org/docs/context.html#contextprovider) to enable device-specific UI for Gestalt components that support it.
 */
export default function DeviceTypeProvider({ children, deviceType }: Props): Node {
  return <DeviceTypeContext.Provider value={deviceType}>{children}</DeviceTypeContext.Provider>;
}

export function useDeviceType(): DeviceType {
  return useContext(DeviceTypeContext) ?? defaultDeviceType;
}
