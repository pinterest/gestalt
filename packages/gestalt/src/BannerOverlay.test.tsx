import { create } from 'react-test-renderer';
import BannerOverlay from './BannerOverlay';
import DeviceTypeProvider from './contexts/DeviceTypeProvider';
import Image from './Image';
import Link from './Link';
import Text from './Text';

describe('BannerOverlay', () => {
  test('renders Text only', () => {
    const tree = create(
      <BannerOverlay
        message="Same great profile, slightly new look. Learn more?"
        title="Profile"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders in desktop device', () => {
    const tree = create(
      <DeviceTypeProvider deviceType="desktop">
        <BannerOverlay
          message="A description"
          offset={{ bottom: 10, top: 15 }}
          title="Text and Image"
        />
      </DeviceTypeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders in mobile device', () => {
    const tree = create(
      <DeviceTypeProvider deviceType="mobile">
        <BannerOverlay
          message="A description"
          offset={{ bottom: 10, top: 15 }}
          title="Text and Image"
        />
      </DeviceTypeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders in mobile device with reversed offset', () => {
    const tree = create(
      <DeviceTypeProvider deviceType="mobile">
        <BannerOverlay
          message="A description"
          offset={{ bottom: 10, top: 15, reverseOffset: true }}
          title="Text and Image"
        />
      </DeviceTypeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders Text + Image', () => {
    const tree = create(
      <BannerOverlay
        message={
          <Text inline weight="bold">
            Saved to{' '}
            <Link
              display="inlineBlock"
              href="https://www.pinterest.com/search/pins/?q=home%20decor"
              underline="hover"
            >
              Home decor
            </Link>
          </Text>
        }
        thumbnail={{
          image: (
            <Image
              alt=""
              naturalHeight={1}
              naturalWidth={1}
              src="https://i.pinimg.com/474x/b2/55/ed/b255edbf773ffb3985394e6efb9d2a49.jpg"
            />
          ),
        }}
        title="Text and Image"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders Text + Image + Button', () => {
    const tree = create(
      <BannerOverlay
        message={
          <Text inline weight="bold">
            Saved to{' '}
            <Link
              display="inlineBlock"
              href="https://www.pinterest.com/search/pins/?q=home%20decor"
              underline="hover"
            >
              Home decor
            </Link>
          </Text>
        }
        thumbnail={{
          image: (
            <Image
              alt=""
              naturalHeight={1}
              naturalWidth={1}
              src="https://i.pinimg.com/474x/b2/55/ed/b255edbf773ffb3985394e6efb9d2a49.jpg"
            />
          ),
        }}
        title="Text, image and button"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
