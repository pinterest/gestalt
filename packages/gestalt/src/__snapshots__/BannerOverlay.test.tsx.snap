// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerOverlay renders Text + Image + Button 1`] = `
Array [
  <div
    className="box smDisplayFlex xsDisplayNone"
  >
    <div
      className="box contentCenter default fit fixed justifyBetween paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionRow xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 900,
          "position": "fixed",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem selfCenter"
        >
          <div
            aria-hidden={true}
            className="box"
          >
            <div
              className="Mask rounding2 willChangeTransform"
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <div
                className="box relative"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "paddingBottom": "100%",
                  }
                }
              >
                <img
                  alt=""
                  className="img"
                  fetchpriority="auto"
                  loading="auto"
                  onError={[Function]}
                  onLoad={[Function]}
                  src="https://i.pinimg.com/474x/b2/55/ed/b255edbf773ffb3985394e6efb9d2a49.jpg"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text, image and button
            </div>
          </div>
          <span
            className="textColorOverrideLight"
          >
            <span
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Saved to
               
              <a
                className="noOutline inheritColor rounding0 inlineBlock tapTransition hoverUnderline noUnderline outlineFocus"
                href="https://www.pinterest.com/search/pins/?q=home%20decor"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                rel=""
              >
                Home decor
              </a>
            </span>
          </span>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfCenter xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="box smDisplayNone xsDisplayFlex"
  >
    <div
      className="box default fit fixed justifyCenter paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionColumn xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 348,
          "position": "fixed",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem selfBaseline"
        >
          <div
            aria-hidden={true}
            className="box"
          >
            <div
              className="Mask rounding2 willChangeTransform"
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <div
                className="box relative"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "paddingBottom": "100%",
                  }
                }
              >
                <img
                  alt=""
                  className="img"
                  fetchpriority="auto"
                  loading="auto"
                  onError={[Function]}
                  onLoad={[Function]}
                  src="https://i.pinimg.com/474x/b2/55/ed/b255edbf773ffb3985394e6efb9d2a49.jpg"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text, image and button
            </div>
          </div>
          <div
            className="box marginBottom2"
          >
            <span
              className="textColorOverrideLight"
            >
              <span
                className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
              >
                Saved to
                 
                <a
                  className="noOutline inheritColor rounding0 inlineBlock tapTransition hoverUnderline noUnderline outlineFocus"
                  href="https://www.pinterest.com/search/pins/?q=home%20decor"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  rel=""
                >
                  Home decor
                </a>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfEnd xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`BannerOverlay renders Text + Image 1`] = `
Array [
  <div
    className="box smDisplayFlex xsDisplayNone"
  >
    <div
      className="box contentCenter default fit fixed justifyBetween paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionRow xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 900,
          "position": "fixed",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem selfCenter"
        >
          <div
            aria-hidden={true}
            className="box"
          >
            <div
              className="Mask rounding2 willChangeTransform"
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <div
                className="box relative"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "paddingBottom": "100%",
                  }
                }
              >
                <img
                  alt=""
                  className="img"
                  fetchpriority="auto"
                  loading="auto"
                  onError={[Function]}
                  onLoad={[Function]}
                  src="https://i.pinimg.com/474x/b2/55/ed/b255edbf773ffb3985394e6efb9d2a49.jpg"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text and Image
            </div>
          </div>
          <span
            className="textColorOverrideLight"
          >
            <span
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Saved to
               
              <a
                className="noOutline inheritColor rounding0 inlineBlock tapTransition hoverUnderline noUnderline outlineFocus"
                href="https://www.pinterest.com/search/pins/?q=home%20decor"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                rel=""
              >
                Home decor
              </a>
            </span>
          </span>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfCenter xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="box smDisplayNone xsDisplayFlex"
  >
    <div
      className="box default fit fixed justifyCenter paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionColumn xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 348,
          "position": "fixed",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem selfBaseline"
        >
          <div
            aria-hidden={true}
            className="box"
          >
            <div
              className="Mask rounding2 willChangeTransform"
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <div
                className="box relative"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "paddingBottom": "100%",
                  }
                }
              >
                <img
                  alt=""
                  className="img"
                  fetchpriority="auto"
                  loading="auto"
                  onError={[Function]}
                  onLoad={[Function]}
                  src="https://i.pinimg.com/474x/b2/55/ed/b255edbf773ffb3985394e6efb9d2a49.jpg"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text and Image
            </div>
          </div>
          <div
            className="box marginBottom2"
          >
            <span
              className="textColorOverrideLight"
            >
              <span
                className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
              >
                Saved to
                 
                <a
                  className="noOutline inheritColor rounding0 inlineBlock tapTransition hoverUnderline noUnderline outlineFocus"
                  href="https://www.pinterest.com/search/pins/?q=home%20decor"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  rel=""
                >
                  Home decor
                </a>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfEnd xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`BannerOverlay renders Text only 1`] = `
Array [
  <div
    className="box smDisplayFlex xsDisplayNone"
  >
    <div
      className="box contentCenter default fit fixed justifyBetween paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionRow xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 900,
          "position": "fixed",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Profile
            </div>
          </div>
          <span
            className="default alignStart breakWord lineClamp Text fontSize300 fontWeightNormal"
            style={
              Object {
                "WebkitLineClamp": 2,
              }
            }
          >
            Same great profile, slightly new look. Learn more?
          </span>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfCenter xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="box smDisplayNone xsDisplayFlex"
  >
    <div
      className="box default fit fixed justifyCenter paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionColumn xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 348,
          "position": "fixed",
          "top": 0,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Profile
            </div>
          </div>
          <div
            className="box marginBottom2"
          >
            <span
              className="default alignStart breakWord lineClamp Text fontSize300 fontWeightNormal"
              style={
                Object {
                  "WebkitLineClamp": 2,
                }
              }
            >
              Same great profile, slightly new look. Learn more?
            </span>
          </div>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfEnd xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`BannerOverlay renders in desktop device 1`] = `
Array [
  <div
    className="box smDisplayFlex xsDisplayNone"
  >
    <div
      className="box contentCenter default fit fixed justifyBetween paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionRow xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 900,
          "position": "fixed",
          "top": 15,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text and Image
            </div>
          </div>
          <span
            className="default alignStart breakWord lineClamp Text fontSize300 fontWeightNormal"
            style={
              Object {
                "WebkitLineClamp": 2,
              }
            }
          >
            A description
          </span>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfCenter xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="box smDisplayNone xsDisplayFlex"
  >
    <div
      className="box default fit fixed justifyCenter paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionColumn xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 348,
          "position": "fixed",
          "top": 15,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text and Image
            </div>
          </div>
          <div
            className="box marginBottom2"
          >
            <span
              className="default alignStart breakWord lineClamp Text fontSize300 fontWeightNormal"
              style={
                Object {
                  "WebkitLineClamp": 2,
                }
              }
            >
              A description
            </span>
          </div>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfEnd xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`BannerOverlay renders in mobile device 1`] = `
Array [
  <div
    className="box smDisplayFlex xsDisplayNone"
  >
    <div
      className="box contentCenter default fit fixed justifyBetween paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionRow xsDisplayFlex"
      style={
        Object {
          "bottom": 10,
          "left": "50%",
          "maxWidth": 900,
          "position": "fixed",
          "top": "unset",
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text and Image
            </div>
          </div>
          <span
            className="default alignStart breakWord lineClamp Text fontSize300 fontWeightNormal"
            style={
              Object {
                "WebkitLineClamp": 2,
              }
            }
          >
            A description
          </span>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfCenter xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="box smDisplayNone xsDisplayFlex"
  >
    <div
      className="box default fit fixed justifyCenter paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionColumn xsDisplayFlex"
      style={
        Object {
          "bottom": 10,
          "left": "50%",
          "maxWidth": 348,
          "position": "fixed",
          "top": "unset",
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text and Image
            </div>
          </div>
          <div
            className="box marginBottom2"
          >
            <span
              className="default alignStart breakWord lineClamp Text fontSize300 fontWeightNormal"
              style={
                Object {
                  "WebkitLineClamp": 2,
                }
              }
            >
              A description
            </span>
          </div>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfEnd xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`BannerOverlay renders in mobile device with reversed offset 1`] = `
Array [
  <div
    className="box smDisplayFlex xsDisplayNone"
  >
    <div
      className="box contentCenter default fit fixed justifyBetween paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionRow xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 900,
          "position": "fixed",
          "top": 15,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text and Image
            </div>
          </div>
          <span
            className="default alignStart breakWord lineClamp Text fontSize300 fontWeightNormal"
            style={
              Object {
                "WebkitLineClamp": 2,
              }
            }
          >
            A description
          </span>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfCenter xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="box smDisplayNone xsDisplayFlex"
  >
    <div
      className="box default fit fixed justifyCenter paddingX4 paddingY3 rounding4 shadow smPaddingY4 xsDirectionColumn xsDisplayFlex"
      style={
        Object {
          "bottom": "unset",
          "left": "50%",
          "maxWidth": 348,
          "position": "fixed",
          "top": 15,
          "transform": "translateX(-50%)",
          "width": "calc(100% - 32px)",
        }
      }
    >
      <div
        className="Flex rowGap4 columnGap4 xsDirectionRow xsItemsCenter"
      >
        <div
          className="FlexItem flexGrow"
        >
          <div
            className="Flex rowGap0 columnGap0 justifyBetween xsDirectionRow"
          >
            <div
              className="default alignStart breakWord Text fontSize300 fontWeightSemiBold"
            >
              Text and Image
            </div>
          </div>
          <div
            className="box marginBottom2"
          >
            <span
              className="default alignStart breakWord lineClamp Text fontSize300 fontWeightNormal"
              style={
                Object {
                  "WebkitLineClamp": 2,
                }
              }
            >
              A description
            </span>
          </div>
        </div>
      </div>
      <div
        className="Flex rowGap4 columnGap4 selfEnd xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box flexWrap marginBottomN1 marginEndN1 marginStartN1 marginTopN1 xsDisplayFlex"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
